docs/setup/k8s/running-in-production-k8s.md

```diff
@@ -62,50 +63,65 @@ Below is a quick example of how to setup the a specific stable version of nuclio
         ./hack/k8s/helm/nuclio/
     ```
 
-  See [the helm chart's values file](/hack/k8s/helm/nuclio/values.yaml) for a full list of configurable parameters
+<a id="multi-tenancy"></a>
+## Multi-Tenancy
 
+Implementation of multi-tenancy can be done in many different ways and to various degrees.
+The experience of the Nuclio team has lead to the adoption of the Kubernetes approach of tenant isolation using namespaces.
+Note:
 
-## Multi Tenancy
+- To achieve tenant separation for various Nuclio projects and functions, and to avoid cross-tenant contamination and resource races, a fully functioning Nuclio deployment is used in each namespace and the Nuclio controller is configured to be namespaced.
+  This means that the controller handles Nuclio resources (functions, function events, and projects) only within its own namespace.
+  This is supported by using the `controller.namespace` [Helm values](/hack/k8s/helm/nuclio/values.yaml) configuration.
+- To provide ample separation at the level of the container image, it's highly recommended that the Nuclio deployments of multiple tenants either don't share container registries, or that they don't share a tenant when using a multi-tenant registry (such as `docker.io` or `quay.io`).
 
-Implementation of multi-tenancy can be done in many different forms and to various degrees. Our experience have led us to adopt the k8s approach of tenant isolation using namespaces.
-- To achieve tenant separation for various nuclio projects and functions, and to avoid cross-tenant contamination and resource races, we have opted to deploy in each namespace a fully functioning Nuclio deployment, and configure Nuclio's controller to be namespaced.
-  This means the controller will handle Nuclio resources (functions, function-events, projects) only within its own namespace. This is supported via `Values.controller.namespace` in the helm chart values.  
-- To provide ample separation on the docker image level, we highly recommend that the Nuclio deployments of various tenants will not share docker registries, or will not share a tenant, if a multi-tenant registry is used (like `docker.io` or `quay.io`) 
+<a id="version-freezing"></a>
+## Freezing a qualified version
 
+When working in production, you need reproducibility and consistency.
+It's therefore recommended that you don't use the latest stable version, but rather qualify a specific Nuclio version and "freeze" it in your configuration.
+Stick with this version until you qualify a newer version for your system.
+Because Nuclio adheres to backwards-compatibility standards between patch versions, and even minor version updates don't typically break major functionality, the process of qualifying a newer Nuclio version should typically be short and easy.
```

Comment:
`typically` used twice

Reply:
Nice catch. I changed the second instance to "generally" — see commit bf210de.